source/openqasm/openqasm3/parser.py

```diff
@@ -331,6 +331,20 @@ def visitDefcalStatement(self, ctx: qasm3Parser.DefcalStatementContext):
             self.visit(ctx.returnSignature().scalarType()) if ctx.returnSignature() else None
         )
 
+        defcal_target: qasm3Parser.DefcalTargetContext = ctx.defcalTarget()
+        if defcal_target.DELAY():
+            duration_ident = defcal_target.Identifier()
+            span = get_span(duration_ident)
+            arguments.append(
+                add_span(
+                    ast.ClassicalArgument(
+                        type=add_span(ast.DurationType(), span=span),
+                        name=add_span(ast.Identifier(duration_ident.getText()), span=span)
+                    ),
+                    span=span,
+                )
+            )
```

Comment:
This implicitly overloads the `argument` field of the node in the AST, making
```
defcal delay[t]
```
appear very similar to another symbol that could have been defined as (invalid typing in the argument list aside)
```
defcal my_symbol(duration t)
```

To me, the "gate parameter list" and the "duration specifier" would be better off as two logically separate abstract parts of the syntax: subject to what's decided in the text of the specification, I'd be in favour of adding a `.duration` field to the AST node `CalibrationDefinition` instead of overloading the parameter list like this.